src/az390.java

```diff
@@ -4171,16 +4200,24 @@ private void process_bal_op(){
          get_hex_vreg(3);      // V3
          get_hex_zero(1);  
          skip_comma();
-         get_hex_byte();      // I4
-        	if ((tz390.op_code[bal_op_index].length() == 5)) {
-        		get_hex_op(5,1); // M5
-        	} else {
-				if (!bal_abort && exp_next_char(',')){
-      		       skip_comma();
-      		       get_hex_reg();   // M4
-      	        } else {
-      	     	   get_hex_zero(1);       
-      	        }
+		 get_hex_byte(); 
+		 if ((tz390.op_code[bal_op_index].length() == 5)) {
+			get_hex_op(5,1); // M5
+		 } else {    // I4
+		    if (!bal_abort && exp_next_char(',')){
+			   skip_comma();
+			   get_hex_reg();   // M5
+		    } else {
+			   get_hex_zero(1);    // M5   
+			}
+		 }
+		 if (tz390.op_code[bal_op_index].substring(0,4).equals("E672")     // E672 VRIf VSRPR V1,V2,V3,I4,M5 DSH #414 z16
+		  || tz390.op_code[bal_op_index].substring(0,4).equals("E670")){     // E670 VRIf VPKZR V1,V2,V3,I4,M5 DSH #414 z16
```

Comment:
Suggestion only. Swap the checks. That is, change to "if ("E670" or "E672") ...".